app/models/CategoryModel.ts

```diff
@@ -0,0 +1,88 @@
+import { Repository } from 'typeorm';
+import { Dependencies } from '../Types';
+import Category from '../orm/entities/Category';
+import Boom from 'boom';
+
+class CategoryModel {
+  private categoryRepo: Repository<Category>;
+
+  constructor(opts: Dependencies) {
+    this.categoryRepo = opts.categoryRepo;
+  }
+
+  public getAllCategories = async () => {
+    try {
+      const categories = await this.categoryRepo.find();
+      if (categories.length > 0) {
+        return categories;
+      } else {
+        throw Boom.notFound('Categories not found');
+      }
+    } catch (error) {
+      throw Boom.isBoom(error)
+        ? error
+        : Boom.serverUnavailable('Something wrong with the database connection');
```

Comment:
Hmm I would say `badImplementation` is bit better error code in cases that you don't know that happened. `serverUnavailable` is usually when server is not responding quickly enough